File: Benchmark/Compute/Pi/Pi.java
public class Pi {

    public static double calculatePi(int terms) {
        double pi = 0.0;
        for (int k = 0; k < terms; k++) {
            if (k % 2 == 0) {
                pi += 1.0 / (2 * k + 1);
            } else {
                pi -= 1.0 / (2 * k + 1);
            }
        }
        return 4 * pi;
    }

    public static void main(String[] args) {
        int terms = Integer.parseInt(args[0]);

        double pi = calculatePi(terms);
        System.out.printf("%.12f\n", pi);
    }
}

File: Benchmark/Compute/Pi/Pi.rs
use std::env;

fn calculate_pi(terms: usize) -> f64 {
    let mut pi = 0.0;
    for k in 0..terms {
        if k % 2 == 0 {
            pi += 1.0 / (2 * k + 1) as f64;
        } else {
            pi -= 1.0 / (2 * k + 1) as f64;
        }
    }
    4.0 * pi
}

fn main() {
    let args: Vec<String> = env::args().collect();
    assert!(args.len() == 2, "Usage: pi <terms>");
    let terms = args[1].parse::<usize>().expect("Invalid integer");
    let pi = calculate_pi(terms);
    println!("{:.12}", pi);
}
